benchmarks/cugraph/standalone/cugraph_bulk_sampling.py

```diff
@@ -0,0 +1,596 @@
+    # Required by dask; need to skip dummy partitions.
+    if partition_info is None:
+        return cudf.DataFrame({
+            'batch': cudf.Series(dtype='int32'),
+            'start': cudf.Series(dtype='int64')
+        })
+    
+    partition = partition_info['number']
+    if partition is None:
+        raise ValueError('division is absent')
+
+    num_batches = int(ceil(len(bdf) / batch_size))
+    
+    batch_ids = cupy.repeat(
+        cupy.arange(num_batches * partition, num_batches * (partition + 1), dtype='int32'),
+        batch_size
+    )[:len(bdf)]
+
+    bdf = bdf.reset_index(drop=True)
+    bdf['batch'] = cudf.Series(batch_ids)
+
+    return bdf
+
+
+def _replicate_df(df: cudf.DataFrame, replication_factor: int, col_offsets:Dict[str, int], partition_info: Optional[Union[dict, str]] = None):
+    # Required by dask; need to skip dummy partitions.
+    if partition_info is None:
+        return cudf.DataFrame({
+            'batch': cudf.Series(dtype='int32'),
+            'start': cudf.Series(dtype='int64')
+        })
+    
+    if replication_factor > 1:
+        for r in range(1, replication_factor):
+            df_replicated = df
+            for col, offset in col_offsets.items():
+                df_replicated[col] += offset * r
+        
+            df = cudf.concat([df, df_replicated], ignore_index=True)
+    
+    return df
+
+
+@get_allocation_counts_dask_lazy(return_allocations=True, logging=True)
+def sample_graph(G, label_df, output_path,seed=42, batch_size=500, seeds_per_call=200000, batches_per_partition=100, fanout=[5, 5, 5]):
+    cupy.random.seed(seed)
+
+    sampler = BulkSampler(
+        batch_size=batch_size,
+        output_path=output_path,
+        graph=G,
+        fanout_vals=fanout,
+        with_replacement=False,
+        random_state=seed,
+        seeds_per_call=seeds_per_call,
+        batches_per_partition=batches_per_partition,
+        log_level = 'INFO'
+    )
+
+    from dask.distributed import wait, default_client
+    n_workers = len(default_client().scheduler_info()['workers'])
+
+    meta = cudf.DataFrame({
+        'node': cudf.Series(dtype='int64'),
+        'batch': cudf.Series(dtype='int32')
+    })
+
+    
+    batch_df = label_df.map_partitions(_make_batch_ids, batch_size, n_workers, meta=meta).persist()
+    del label_df
+    print('created batches')
+    
+
+    start_time = perf_counter()
+    sampler.add_batches(batch_df, start_col_name='node', batch_col_name='batch')
+    sampler.flush()
+    end_time = perf_counter()
+    print('flushed all batches')
+    return (end_time - start_time)
+
+
+def load_disk_dataset(dataset, dataset_dir='.', reverse_edges=True, replication_factor=1):
+    path = os.path.join(dataset_dir, dataset)
+    parquet_path = os.path.join(path, 'parquet')
+
+    with open(os.path.join(path, 'meta.json')) as meta_file:
+        meta = json.load(meta_file)
+    
```

Comment:
Can i have an example of `meta.json`. 

I basically need below for DGL dataloaders. https://github.com/rapidsai/cugraph/blob/c49643c1263bb5c0a00a5f9c2c0b995a075a1396/python/cugraph-dgl/cugraph_dgl/dataloading/dataset.py#L83-L87
```python3
num_nodes_dict: Dict[str, int],
etype_id_dict: Dict[int, Tuple[str, str, str]],
etype_offset_dict: Dict[Tuple[str, str, str], int],
ntype_offset_dict: Dict[str, int],
```

Reply:
I can add offsets to the `output_meta.json` if that will make it easier for DGL

Reply:
Output meta example:
```
{
"dataset": "ogbn_papers100M",
"dataset_dir": "/datasets/abarghi",
"seed": 62,
"batch_size": 500,
"seeds_per_call": 500000,
"fanout": [10, 25],
"replication_factor": 2,
"execution_time": 10.963991190772504,
"offsets": {
     "paper": 0
}
}
```

Reply:
In that script I put a comment that it's using the cuGraph-PyG offset conventions

Reply:
For homogenous graphs, I dont need anything but `total_number_of_nodes `


https://github.com/rapidsai/cugraph/blob/c49643c1263bb5c0a00a5f9c2c0b995a075a1396/python/cugraph-dgl/cugraph_dgl/dataloading/dataset.py#L33-L34

Reply:
Which my loader script uses for heterogeneous graphs

Reply:
I can output those offsets in the output_meta.json file so that you can use them.

Reply:
ok, I'll work on that